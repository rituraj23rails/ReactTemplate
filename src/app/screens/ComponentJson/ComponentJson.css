.jsonContainer {
    display: flex;
    justify-content: center;
    background-color: #000;
    height: 100vh;
}

.rawData {
    font-size: 14px;
    color: #fff;
    padding: 50px;
    text-align: left;
}

@media screen and (max-width: 820px) {
    .rawData {
        font-size: 10px;
        color: #fff;
        padding: 50px;
        text-align: left;
    }

}

@media screen and (max-width: 480px) {
    .rawData {
        font-size: 8px;
        color: #fff;
        padding: 50px;
        text-align: left;
    }
}
