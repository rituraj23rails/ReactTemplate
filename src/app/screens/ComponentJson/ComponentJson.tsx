import React, { Component, useRef } from 'react';
import './ComponentJson.css';
import WithRouter from '../../../WithRouter';

export interface Props {
    navigate: any;
    id: string;
    location: any;
}

interface S {
}

interface SS {
    id: any;
}

class ComponentJson extends Component<Props, S, SS> {
    constructor(props: any) {
        super(props);
        this.state = {
        };
    };

    componentDidMount = async () => {
        console.log('@@@ =======', this.props.location.state.rawData);
    }

    componentWillUnmount = () => {

    }

    render() {
        return (
            <div className='jsonContainer'>
                <p className="rawData">{JSON.stringify(this.props.location.state.rawData)}</p>
            </div>
        )
    }
};

export default WithRouter(ComponentJson);
export { ComponentJson };
