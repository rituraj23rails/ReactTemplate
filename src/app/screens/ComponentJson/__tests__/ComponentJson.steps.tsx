import { defineFeature, loadFeature } from "jest-cucumber"
import { shallow, ShallowWrapper, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { ComponentJson } from "../ComponentJson";

configure({adapter: new Adapter()});

const screenProps = {
    navigate: jest.fn(),
    location: {
        state: {
            params: ''
        }
    },
    id: "ComponentJson",
}

const feature = loadFeature('./src/app/screens/ComponentJson/__tests__/features/ComponentJson-scenarios.feature');

defineFeature(feature, (test) => {

    test('User navigates Component JSON Screen', ({ given, when, then }) => {
        let ComponentJsonScreenWrapper: ShallowWrapper;
        let ComponentJsonScreenInstance: ComponentJson;

        given('I am a User loading Component JSON Screen', () => {
            ComponentJsonScreenWrapper = shallow(<ComponentJson {...screenProps} />)
        });

        when('I navigate to the Component JSON Screen', () => {
            ComponentJsonScreenInstance = ComponentJsonScreenWrapper.instance() as ComponentJson
        });

        then('Component JSON Screen will load with out errors', () => {
            ComponentJsonScreenInstance.componentDidMount();
            expect(ComponentJsonScreenWrapper).toBeTruthy()
            expect(ComponentJsonScreenWrapper).toMatchSnapshot()
        });

        then('I can leave the screen with out errors', () => {
            ComponentJsonScreenInstance.componentWillUnmount();
            expect(ComponentJsonScreenWrapper).toBeTruthy()
            expect(ComponentJsonScreenWrapper).toMatchSnapshot()
        });
    });

});
