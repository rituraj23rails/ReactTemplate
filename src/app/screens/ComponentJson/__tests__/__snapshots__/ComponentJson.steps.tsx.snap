// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component JSON Screen User navigates Component JSON Screen 1`] = `
<div
  className="jsonContainer"
>
  <p
    className="rawData"
  />
</div>
`;

exports[`Component JSON Screen User navigates Component JSON Screen 2`] = `
<div
  className="jsonContainer"
>
  <p
    className="rawData"
  />
</div>
`;
